elide-core/src/main/java/com/yahoo/elide/annotation/UserPermission.java

```diff
@@ -0,0 +1,28 @@
+/*
+ * Copyright 2015, Yahoo Inc.
+ * Licensed under the Apache License, Version 2.0
+ * See LICENSE file in project root for terms.
+ */
+package com.yahoo.elide.annotation;
+
+import com.yahoo.elide.optimization.UserCheck;
+
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+/**
+ * Annotation providing a user-level permission set on an entity. This permission takes precedence over all others
+ * and can be used to short-circuit potentially expensive permissions in future check types.
+ */
+@Target({ElementType.TYPE})
+@Retention(RetentionPolicy.RUNTIME)
+public @interface UserPermission {
+    /**
+     * UserCheck's to run before evaluating additional checks on object
+     *
+     * @return
+     */
+    Class<? extends UserCheck>[] value() default {};
```

Comment:
FilterScope logic seems to imply the use of any and all still.  I think it makes sense to have.